### PR TITLE
Mock http server for LLM; Integration test for visualization tool

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description
1. mock LLM
2. integration test for visualization tool
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by xinyual — APPROVED** [review]

File: src/test/java/org/opensearch/integTest/ToolIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+                    + "}"
+            );
+        Response response = executeRequest(request);
+
+        String resp = readResponse(response);
+
+        return JsonParser.parseString(resp).getAsJsonObject().get("model_id").getAsString();
+    }
+
+    private String setupConversationalAgent(String modelId) throws IOException {
```
Comment (xinyual) [inline src/test/java/org/opensearch/integTest/ToolIntegrationTest.java:166]: I think a flow agent is enough to cover tool IT test like https://github.com/opensearch-project/skills/blob/34ae75f81ed47ecc2e4006d32343e12fa388678b/src/test/java/org/opensearch/integTest/NeuralSparseSearchToolIT.java#L94

**Review by xinyual — CHANGES_REQUESTED** [review]

File: src/test/java/org/opensearch/integTest/ToolIntegrationTest.java (hunk 1 above)
Comment (Hailong-am) [reply src/test/java/org/opensearch/integTest/ToolIntegrationTest.java:166]: you are right, conversational agent is used to test mock LLM server. To run integration test for tools, flow agent is enough.

**Review by xinyual — APPROVED** [review]

File: src/test/java/org/opensearch/integTest/PromptHandler.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    boolean apply(String prompt) {
+        return prompt.contains(toolInput().getQuestion());
+    }
+
+    ToolInput toolInput() {
+        return new ToolInput();
+    }
+
+    String response(String prompt) {
+        if (prompt.contains("TOOL RESPONSE: ")) {
```
Comment (zane-neo) [inline src/test/java/org/opensearch/integTest/PromptHandler.java:26]: Will this `TOOL RESPONSE: ` be used when writing actual ITs for a tool? If so, should we extract it to a common constant?

File: src/test/java/org/opensearch/integTest/PromptHandler.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    @Builder
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @Data
+    static class ToolInput {
+        String question;
+        String toolType;
+        String toolInput;
```
Comment (zane-neo) [inline src/test/java/org/opensearch/integTest/PromptHandler.java:51]: Can we change toolType and toolInput to action and actionInput to be consistency?

File: src/test/java/org/opensearch/integTest/ToolIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    public static String readResponse(Response response) throws IOException {
+        StringBuilder sb = new StringBuilder();
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(response.getEntity().getContent()))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                sb.append(line);
+            }
+        }
+        return sb.toString();
```
Comment (zane-neo) [inline src/test/java/org/opensearch/integTest/ToolIntegrationTest.java:224]: Is it possible to use third party method like `IOUtils.toString()` to simplify this code block?

File: src/test/java/org/opensearch/integTest/PromptHandler.java (hunk 2 above)
Comment (Hailong-am) [reply src/test/java/org/opensearch/integTest/PromptHandler.java:26]: That depends. For most tools that use flow agent to do integration test, they will not interact with LLM. This default implementation is used to test conversational agent flow with mock LLM server. Based on this assumption, we can leave it as is.

File: src/test/java/org/opensearch/integTest/PromptHandler.java (hunk 3 above)
Comment (Hailong-am) [reply src/test/java/org/opensearch/integTest/PromptHandler.java:51]: sure, good suggestion.

File: src/test/java/org/opensearch/integTest/ToolIntegrationTest.java (hunk 4 above)
Comment (Hailong-am) [reply src/test/java/org/opensearch/integTest/ToolIntegrationTest.java]: done, update to OpenSearch built-in class instead `org.opensearch.common.io.Streams.readAllLines(ins)`

**Review by zane-neo — APPROVED** [review]

**Pull request merged**